services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/Sort.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management.device;
+
+import java.util.Objects;
+
+import org.eclipse.hono.util.RegistryManagementConstants;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.vertx.core.json.pointer.JsonPointer;
+
+/**
+ * It specifies properties to sort the result set during search operation
+ * in Device Registry Management API.
+ */
+public final class Sort {
+
+    @JsonProperty(RegistryManagementConstants.FIELD_FILTER_FIELD)
+    private JsonPointer field;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_SORT_DIRECTION)
+    private DIRECTION direction = DIRECTION.asc;
+
+    /**
+     * An enum defining the sort directions.
+     */
+    public enum DIRECTION {
+        asc,
+        desc
+    }
+
```

Comment:
can we add a constructor that accepts the mandatory field property and remove the *setField* method and make field final?